ui/app/src/components/shared/airlock/AirlockNewRequest.tsx

```diff
@@ -0,0 +1,219 @@
+import { DefaultButton, Dialog, DialogFooter, DocumentCard, DocumentCardDetails, DocumentCardPreview, DocumentCardTitle, DocumentCardType, getTheme, Icon, IDocumentCardPreviewProps, IStackTokens, Panel, PanelType, PrimaryButton, Spinner, SpinnerSize, Stack, TextField } from "@fluentui/react";
+import { useCallback, useContext, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { WorkspaceContext } from "../../../contexts/WorkspaceContext";
+import { HttpMethod, useAuthApiCall } from "../../../hooks/useAuthApiCall";
+import { AirlockRequest, AirlockRequestType, NewAirlockRequest } from "../../../models/airlock";
+import { ApiEndpoint } from "../../../models/apiEndpoints";
+import { APIError } from "../../../models/exceptions";
+import { ExceptionLayout } from "../ExceptionLayout";
+
+interface AirlockNewRequestProps {
+  onCreateRequest: (request: AirlockRequest) => void;
+}
+
+const stackTokens: IStackTokens = { childrenGap: 20 };
+const { palette, fonts } = getTheme();
+
+const importPreviewGraphic: IDocumentCardPreviewProps = {
```

Comment:
sort of a `nit` - can we move all the boilerplate / layout code to the bottom, as we are less likely to care about it?

Reply:
Good idea